component/discovery/consulagent/consulagent.go

```diff
@@ -0,0 +1,85 @@
+package consulagent
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/river/rivertypes"
+	promcfg "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.consulagent",
+		Args:    Arguments{},
+		Exports: discovery.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Arguments struct {
+	Server          string            `river:"server,attr,optional"`
+	Token           rivertypes.Secret `river:"token,attr,optional"`
+	Datacenter      string            `river:"datacenter,attr,optional"`
+	TagSeparator    string            `river:"tag_separator,attr,optional"`
+	Scheme          string            `river:"scheme,attr,optional"`
+	Username        string            `river:"username,attr,optional"`
+	Password        rivertypes.Secret `river:"password,attr,optional"`
+	RefreshInterval time.Duration     `river:"refresh_interval,attr,optional"`
+	Services        []string          `river:"services,attr,optional"`
+	ServiceTags     []string          `river:"tags,attr,optional"`
+	TLSConfig       config.TLSConfig  `river:"tls_config,block,optional"`
```

Comment:
Do we also need to configure the `node_meta` field that exist in the [upstream config](https://grafana.com/docs/loki/v2.8.x/clients/promtail/configuration/)?

Reply:
like allow_stale, node_meta is not needed for the consul agent, it a copy paste mistake from the prometheus code that they made in promtail. I removed in the agent because the field was not used

Reply:
I will put a warning for the node_meta. The allow_stale is not in the doc of promtail and also it's a bool which makes it a bit awkward to check whether it was set or not so maybe I don't put a warning for this one?

Reply:
Yep, I'm okay with this.